Sources/KituraNet/ClientRequest.swift

```diff
@@ -550,8 +551,20 @@ public class ClientRequest {
             initializeClientBootstrap(eventLoopGroup: group)
         }
 
-        let hostName = URL(string: percentEncodedURL)?.host ?? "" //TODO: what could be the failure path here
-        let portNumber = URL(string: percentEncodedURL)?.port ?? 8080
+        let hostName = percentEncodedURL.host! //TODO: what could be the failure path here
+        let portNumber: Int
+        if let port = percentEncodedURL.port {
+            portNumber = port
+        } else {
+            if percentEncodedURL.scheme == "https" {
+                portNumber = 443
+            } else if percentEncodedURL.scheme == "http" {
+                portNumber = 80
+            } else {
+                portNumber = 8080
+                Log.error("Unknown port for url: \(url)  Assuming 8080")
+            }
+        }
         if self.headers["Host"] == nil {
            let isNotDefaultPort = (portNumber != 443 && portNumber != 80) //Check whether port is not 443/80
```

Comment:
This whole play of port literals gave me an idea:

We could introduce a small harmless enum in one of the Kitura base modules which would encapsulate the typical ports and give names to them?

The bigger benefit is that it could open the door for future additional features to port handling without change risks.

@dannys42 && @mbarnach what do you think?

Reply:
I like the idea: it makes it more obvious (like which port is the default for Kitura), but also keep tracks of standard and custom ports in a robust fashion.

Reply:
Hmm.. you mean an enum equivalent to the /etc/services file?  Might be interesting, but not sure it belongs in Kitura since Kitura really only deals with web traffic?

Reply:
The way I understand it is: instead of 80, you have something like `.http` and instead of 443, `.https`. For 8080, it would be `.default` (names are poorly chosen!). It could also be extended to use `.localhost` and `.domain(let custom)` for the hostnames. Feature wise, the same, but better to add semantic and group the value in a single place. As part of the port enum, the `var isDefaultPort: Bool { self == .http || self == .https }` could be introduced.